src/QsCompiler/Compiler/CompilationLoader.cs

```diff
@@ -631,11 +638,7 @@ private IEnumerable<(LoadedStep, Action<Status>)> InternalRewriteSteps(Uri execu
                 // TODO: It would be nicer to trim unused intrinsics. Currently, this is not possible due to how the
                 // old setup of the C# runtime works. With the new setup (interface-based approach for target
                 // packages), it is possible to trim unused intrinsics.
-                (
-                    new SyntaxTreeTrimming(keepAllIntrinsics: true, dependencies),
-                    this.config.IsExecutable && !this.config.SkipSyntaxTreeTrimming,
-                    s => status.TreeTrimming = s),
-
+                (new SyntaxTreeTrimming(keepAllIntrinsics: true, dependencies), this.config.TrimTree, s => status.TreeTrimming = s),

```

Comment:
Food for thought in the future: per the comment, the unused intrinsics are only needed for C# that includes generated abstract intrinsics. In other words, an intrinsic can be safely trimmed if it is from a target package and is unused. Not needed for this change, but an interesting optimization to consider in the future (perhaps an update to the comment with our latest understandings).

Reply:
Hm, that was meant to be captured by the second half of the comment: "With the new setup (interface-based approach for target packages), it is possible to trim unused intrinsics." I edited the second half in an attempt to make it clearer: Intrinsics that come from packages that use the new interface-based approach (which is the case for target packages) can be trimmed if they are unused. Honestly, I was hoping at some point we would get to a place where everything uses interfaces...